source/math/FloatExtension.ooc

```diff
@@ -126,17 +126,17 @@ extend Float {
 		}
 		result
 	}
-	getRadix: static func (value: Float, valueDigits: Int) -> Float {
+	getRadix: static func (value: This, valueDigits: Int) -> This {
 		(tempValue, result) := This decomposeToCoefficientAndRadix(value, valueDigits)
 		result
 	}
-	getScientificPowerString: static func (value: Float) -> String {
+	getScientificPowerString: static func (value: This) -> String {
 		(coefficient, radix) := This decomposeToCoefficientAndRadix(value, 1)
 		power := log10(radix) as Int
 		result := ""
-		if (coefficient != 1.0f)
-			result = result & coefficient toString() & "×" clone()
-		result = result & "10^" clone() & power toString()
+		if (This absolute(coefficient - 1.0f) > This epsilon)
+			result = result & coefficient toString() >> "×"
+		result = result >> "10^" & power toString()
```

Comment:
And if you want to keep `10`, there should really be a space surrounding the `×`, and the `^exponent` should be replaced with `<sup>exponent</sup`.